### PR TITLE
[BugFix] BE crashed in left_bytes_string

### PR DESCRIPTION
**Pull request by rickif**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/StarRocksTest/issues/1471

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

- [x] I have checked the version labels which the pr will be auto backported to target branch

### CODE REVIEW

File: be/src/exec/vectorized/json_parser.cpp [hunk 1]
```diff
@@ -141,7 +147,22 @@ Status JsonArrayParser::advance() noexcept {
 }
 
 std::string JsonArrayParser::left_bytes_string(size_t sz) noexcept {
```
Comment (chaoyli) [inline be/src/exec/vectorized/json_parser.cpp:149]: You need to add some test cases

**Comment by rickif** [comment]
run all

File: be/src/exec/vectorized/json_parser.cpp (hunk 1 above)
Comment (rickif) [reply be/src/exec/vectorized/json_parser.cpp:149]: Actually, there're unit tests for this method.
The crash is led by `return std::string(_doc.current_location()).substr(0, sz);`, which would access illegal memory when the `_doc` content is not terminated with `\0`. But the implement of std:;string would append `\0` to the end.

**Review by meegoo — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**